From 77f1a1314a267c9f5b7a395344cfb3af57dcd3d6 Mon Sep 17 00:00:00 2001
From: Noriko Hosoi <nhosoi@redhat.com>
Date: Thu, 28 Mar 2013 15:30:13 -0700
Subject: [PATCH] Ticket 623 - cleanAllRUV task fails to cleanup config upon
 completion

Bug description: coverity reported "13158 - Logically dead code'

Fix description: csn_as_string returns allocated csn string,
which needs to be reassigned to csnstr.

https://fedorahosted.org/389/ticket/623

Reviewed by Nathan (Thank you!!)
(cherry picked from commit d5fe6ba773ef2655845df5152901fd5bb6a9d4e9)
---
 ldap/servers/plugins/replication/repl5_replica_config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/plugins/replication/repl5_replica_config.c b/ldap/servers/plugins/replication/repl5_replica_config.c
index 8710c04..0e29c6f 100644
--- a/ldap/servers/plugins/replication/repl5_replica_config.c
+++ b/ldap/servers/plugins/replication/repl5_replica_config.c
@@ -2367,7 +2367,7 @@ delete_cleaned_rid_config(cleanruv_data *clean_data)
     /*
      *  If there is no maxcsn, set the proper csnstr
      */
-    csn_as_string(clean_data->maxcsn, PR_FALSE, csnstr);
+    csnstr = csn_as_string(clean_data->maxcsn, PR_FALSE, csnstr);
     if(csnstr == NULL || csn_get_replicaid(clean_data->maxcsn) == 0){
         csnstr = slapi_ch_strdup("00000000000000000000");
     }
-- 
1.8.1.4

